### PR TITLE
Use nGraph logical And and Or operations

### PR DESCRIPTION
**Pull request by arogowie-intel**
Depends on #127  and #126

### CODE REVIEW

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 1]
```diff
@@ -462,34 +462,27 @@ def Greater(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> N
 def And(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> NgraphNode
     """Perform the `and` logical operation elementwise on two input tensors."""
     left, right = broadcast_for_binary_operation(onnx_node, ng_inputs)
-    left = ng.convert(ng.not_equal(left, 0), int)
-    right = ng.convert(ng.not_equal(right, 0), int)
-    return ng.convert(left * right, bool)
+    return ng.logical_and(ng.convert(left, bool), ng.convert(right, bool))
```
Comment (tsocha) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: ONNX documentation specify input type as bool.
Why do we have to convert inputs to bool?

Comment (arogowie-intel) [reply ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: That's right onnx documentation says it is bool. However in our tests we passed integers which caused errors :) I've not checked doc so it's my mistake.

**Comment by arogowie-intel** [comment]
TESTNOW

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 2]
```diff
@@ -462,34 +464,25 @@ def Greater(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> N
...
+    return ng.logical_and(left, right)
 
 
 def Or(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> NgraphNode
     """Perform the `or` logical operation elementwise on two input tensors."""
     left, right = broadcast_for_binary_operation(onnx_node, ng_inputs)
-    left = ng.convert(ng.not_equal(left, 0), int)
-    right = ng.convert(ng.not_equal(right, 0), int)
-    return (left + right) > 0
+    return ng.logical_or(left, right)
```
Comment (postrational) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: Can we use `ng.convert` to `bool` when the incoming type is not a `bool`?
We could support more cases then. We can also raise a warning.

File: ngraph_onnx/onnx_importer/ops_bridge/opset_01.py [hunk 3]
```diff
@@ -462,34 +464,25 @@ def Greater(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> N
...
 def Xor(onnx_node, ng_inputs):  # type: (NodeWrapper, List[NgraphNode]) -> NgraphNode
     """Perform the `xor` logical operation elementwise on two input tensors."""
     left, right = broadcast_for_binary_operation(onnx_node, ng_inputs)
-    not_left = ng.convert(ng.equal(left, 0), int)
-    left = ng.convert(ng.not_equal(left, 0), int)
-    right = ng.convert(ng.not_equal(right, 0), int)
-    not_right = ng.convert(ng.equal(right, 0), int)
-
-    return ((not_left * right) + (not_right * left)) > 0
+    return ng.logical_or(ng.logical_and(left, ng.logical_not(right)),
```
Comment (postrational) [inline ngraph_onnx/onnx_importer/ops_bridge/opset_01.py]: We could also use `ng.convert` here.

**Pull request merged**